File: Ransom Note.cc
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> NoteHashTable(26, 0), MagHashTable(26, 0);
        
        for (char c: ransomNote) NoteHashTable[int(c - 'a')] ++;
        
        for (char c: magazine) MagHashTable[int(c - 'a')] ++;
        
        for (int i = 0; i < 26; i ++) if (MagHashTable[i] < NoteHashTable[i]) return false;
        
        return true;
    }
};

File: Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) 
    {
        int[] countArr = new int[26];
        for(int i = 0 ; i<magazine.length(); i++)
        {   char ch = magazine.charAt(i);
            countArr[ch-'a']++;
        }
        
        for(int i = 0 ; i<ransomNote.length(); i++)
        {   char ch = ransomNote.charAt(i);
            if(--countArr[ch-'a']<0) return false ;
        }
        return true;
    }
}